src/gwt/src/org/rstudio/studio/client/rsconnect/ui/NewRSConnectAuthPage.java

```diff
@@ -305,14 +305,17 @@ public void execute(NewRSConnectAccountResult input)
                   contents_.showWaiting();
                   
                   // prepare a new window with the auth URL loaded
-                  NewWindowOptions options = new NewWindowOptions();
-                  options.setName(AUTH_WINDOW_NAME);
-                  options.setAllowExternalNavigation(true);
-                  options.setShowDesktopToolbar(false);
-                  display_.openWebMinimalWindow(
-                        result_.getPreAuthToken().getClaimUrl(),
-                        false, 
-                        700, 800, options);
+                  if (canSpawnAuthenticationWindow())
```

Comment:
Yeah I think that makes good sense.

On Tue, Oct 27, 2015 at 11:51 AM, Jonathan <notifications@github.com> wrote:

> In
> src/gwt/src/org/rstudio/studio/client/rsconnect/ui/NewRSConnectAuthPage.java
> <https://github.com/rstudio/rstudio/pull/518#discussion_r43141282>:
>
> > @@ -305,14 +305,17 @@ public void execute(NewRSConnectAccountResult input)
> >                    contents_.showWaiting();
> >
> >                    // prepare a new window with the auth URL loaded
> > -                  NewWindowOptions options = new NewWindowOptions();
> > -                  options.setName(AUTH_WINDOW_NAME);
> > -                  options.setAllowExternalNavigation(true);
> > -                  options.setShowDesktopToolbar(false);
> > -                  display_.openWebMinimalWindow(
> > -                        result_.getPreAuthToken().getClaimUrl(),
> > -                        false,
> > -                        700, 800, options);
> > +                  if (canSpawnAuthenticationWindow())
>
> Maybe instead of changing the UI we should just call
> Desktop.getFrame().browseUrl(...)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/518/files#r43141282>.
>